code/modules/shuttle/mini_dropship.dm

```diff
@@ -279,10 +280,13 @@
 	if(!origin.placeLandingSpot(target))
 		to_chat(owner, span_warning("You cannot land here."))
 		return
+	if(is_ground_level(origin.z)) //Safety check to prevent instant transmission
+		to_chat(owner, span_warning("The shuttle can't move while docked on the planet"))
+		return
 	origin.shuttle_port.callTime = SHUTTLE_LANDING_CALLTIME
 	origin.next_fly_state = SHUTTLE_ON_GROUND
 	origin.open_prompt = FALSE
-	origin.clean_ui_user()
+	origin.ui_close(C)
```

Comment:
```suggestion
	SStgui.close_user_uis(C, origin)
```

Reply:
That line gets ran in the proc

Reply:
Again, so why is this line necessary at all then?

Reply:
`/**
 * public
 *
 * Called on a UI's object when the UI is closed, not to be confused with
 * client/verb/uiclose(), which closes the ui window
 */
/datum/proc/ui_close(mob/user)
	SIGNAL_HANDLER
`